pkg/asset/installconfig/azure/validation.go

```diff
@@ -17,15 +18,95 @@ import (
 	aztypes "github.com/openshift/installer/pkg/types/azure"
 )
 
+type resourceRequirements struct {
+	minimumVCpus  int64
+	minimumMemory int64
+}
+
+var controlPlaneReq = resourceRequirements{
+	minimumVCpus:  4,
+	minimumMemory: 16,
+}
+
+var computeReq = resourceRequirements{
+	minimumVCpus:  2,
+	minimumMemory: 8,
+}
+
 // Validate executes platform-specific validation.
 func Validate(client API, ic *types.InstallConfig) error {
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, validateNetworks(client, ic.Azure, ic.Networking.MachineNetwork, field.NewPath("platform").Child("azure"))...)
 	allErrs = append(allErrs, validateRegion(client, field.NewPath("platform").Child("azure").Child("region"), ic.Azure)...)
+	allErrs = append(allErrs, validateInstanceTypes(client, ic)...)
 	return allErrs.ToAggregate()
 }
 
+// ValidateInstanceType ensures the instance type has sufficient Vcpu and Memory.
+func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceType string, req resourceRequirements) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	typeMeta, err := client.GetVirtualMachineSku(context.TODO(), instanceType, region)
+	if err != nil {
+		return append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
+	}
+
+	for _, capability := range *typeMeta.Capabilities {
+
+		if strings.EqualFold(*capability.Name, "vCPUs") {
```

Comment:
Are we willing to accept a SKU that does not have the "vCPUs" or "MemoryGB" capability?

Reply:
Currently, if the capability is not present, then we won't show any error. We filter out any SKU that is not of type "virtualMachines". I am unable to find anything in the docs where this type of SKU is guaranteed to have these capabilities specified (or not).

Reply:
I think we should move forward as-is. We can always make changes if we find cu issues later on.